LongestPole/Main.lean

```diff
@@ -0,0 +1,184 @@
+/-
+Copyright (c) 2023 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import Mathlib.Util.Imports
+import Mathlib.Data.String.Defs
+import Std.Lean.Util.Path
+import Cli
+
+/-!
+# `lake exe pole`
+
+Longest pole analysis for Mathlib build times.
+-/
+
+set_option autoImplicit true
+
+open Cli
+
+open Lean Meta
+
+open Lean Core System
+
+/-- Runs a terminal command and retrieves its output -/
+def runCmd (cmd : String) (args : Array String) (throwFailure := true) : IO String := do
+  let out ← IO.Process.output { cmd := cmd, args := args }
+  if out.exitCode != 0 && throwFailure then throw $ IO.userError out.stderr
+  else return out.stdout
+
+def runCurl (args : Array String) (throwFailure := true) : IO String := do
+  runCmd "curl" args throwFailure
+
+def speedCenterRunJson (hash : String) : IO String :=
+  runCurl #["http://speed.lean-fro.org/mathlib4/api/run/e7b27246-a3e6-496a-b552-ff4b45c7236e?hash=" ++ hash]
```

Comment:
What does the string "e7b27246-a3e6-496a-b552-ff4b45c7236e" signify? Should it be hard-coded? I'm fine with that, but may add a short comment explaining this.

Reply:
Good point. It is the `repo_id` for Mathlib.